### PR TITLE
Add go vet to CI pipeline and fix reported issues

### PR DESCRIPTION
**Pull request by AndersonQ**

### CODE REVIEW

**Review by pchila — COMMENTED** [review]
Change looks fine, I am worried about the test failure though...

File: pkg/client/unit_test.go [hunk 1]
```diff
@@ -79,8 +79,7 @@ func TestUnitUpdateState(t *testing.T) {
 }
 
 func TestUnitUpdateAPMConfig(t *testing.T) {
-
-	var initialUnitNoAPM = Unit{
+	initialUnitNoAPM := &Unit{
```
Comment (pchila) [inline pkg/client/unit_test.go:82]: This struct is now shared among different testcases which makes the test fail when all the testcases are run. Just tried to nest it at the cost of some duplication but that take care of the failures without reintroducing the `go vet` violations

Comment (pchila) [reply pkg/client/unit_test.go]: Added a new commit where we create a new unit per testcase

Comment (AndersonQ) [reply pkg/client/unit_test.go]: I was pushing a fix as well, just copying and pasting it all over. Yours is more elegant. Thanks

**Pull request merged**